File: Java/eclipse-workspace/Prj03_Contatore/src/autosalone/Automobile.java
package autosalone;

public class Automobile {

	String marca;
	String modello;
	String colore;
	double prezzo;
	int km;
	
	public Automobile(String marca, String modello, String colore, double prezzo, int km) {
		this.marca = marca;
		this.modello = modello;
		this.colore = colore;
		this.prezzo = prezzo;
		this.km = km;
	}

	@Override
	public String toString() {
		return "Automobile [marca=" + marca + ", modello=" + modello + ", colore=" + colore + ", prezzo=" + prezzo
				+ ", km=" + km + "]";
	}
	
	
	
	
}

File: Php/ISD2025/L08/src/entities/Automobile.php
<?php

class Automobile implements JsonSerializable{

    private $id;
    private $marca;
    private $modello;
    private $cilindrata;
    private $posti;
    private $prezzo;

    public function __get($nomeVariabile) {
       return $this->$nomeVariabile; 
    }

    public function __set($nomeVariabile, $valoreVariabile){
        $this->$nomeVariabile = $valoreVariabile;
    }

    public function jsonSerialize(){
        return [
            'marca'=>$this->marca,
            'modello'=>$this->modello,
            'prezzo'=> number_format( $this->prezzo * 1.1, 2, ',','.' ),
        ];
    }

}

